ompi/mpi/c/recv.c

```diff
@@ -70,6 +71,23 @@ int MPI_Recv(void *buf, int count, MPI_Datatype type, int source,
         OMPI_ERRHANDLER_CHECK(rc, comm, rc, FUNC_NAME);
     }
 
+#if OPAL_ENABLE_FT_MPI
```

Comment:
This is a different behavior from the other receive functions. We need to have a consistent behavior everywhere. Plus this might not be correct if there are pending short messages already received by the PML.

What is ompi_comm_iface_p2p_check_proc really checking ? That the comm has been revoked by the lower level infrastructure or that the error has been already reported upstream ?

Reply:
It checks if the comm has been revoked, or if the target proc is dead, or if any_source is disabled when the target is any_source; All of this is kinda duplicating the work we do in wait in any case. 

It also spins progress once to make sure that we do propagate revoke messages (otherwise they could get stuck forever as no progress would take place in some cases). 

The difference with other Recv calls comes from the fact that his function is blocking when other calls are non-blocking. 

That is, except for mrecv, which is blocking, and still does the full PML handover. I can't remember why it is different.

Reply:
First it would be good if similar functions have similar behavior/constraints. Second, with this implementation it seems as if MPI_Recv iwll quickly fail even when the comm was only internally marked as revoked, while the irecv has a change to do a matching with a possible pending message. Do we deliver the unexpected messages after a fault or a revoke was called or we discard them eagerly ?

Reply:
If the comm is 'internally' revoked, the pending queue will have been cleared before the iRECV posts (that is, if it had caused a RECV posted at the same 'time' to return REVOKED). 

We do clear the pending unmatched frags first order of business and we hold the matching lock  while we are doing so.